src/main/java/io/axoniq/axonserver/connector/command/impl/CommandChannelImpl.java

```diff
@@ -156,7 +157,13 @@ private void handleAck(CommandProviderInbound message, ReplyChannel<CommandProvi
     }
 
     @Override
-    public synchronized void connect() {
+    public void connect() {
+        if (!commandHandlers.isEmpty()) {
```

Comment:
I fail to see where the confusion would be: ignore a connect request if the collection of command handlers is empty.

Reply:
Isn't this channel also used to dispatch commands on? Or would that only be allowed when there are handlers present? Or is that covered upon dispatching of a command, by validating if the connection was already made?

I am not stating it's unclear, I however feel there are several ways around it which in a years time likely don't make this decision overly clear anymore. Yes there's git commit messages, but if a named function describes the intent too I'd personally opt for that. 

If I am making wrong assumptions on the usages of the `CommandChannelImpl`, just let me know.

Reply:
Verbally discussed and resolved; misconception from my end, **and** already covered in the [axon-server-api](https://github.com/AxonIQ/axon-server-api). Hence there's no need to rehash those specifics here.

Reply:
Connect refers to setting up the command stream (in this case), as defined in the CommandService gRPC definition. See https://github.com/AxonIQ/axon-server-api/blob/master/src/main/proto/command.proto#L10. Since this connector is "only" implementing that specification. I would expect anyone picking this up in the future to take a dive into the API specification before diving into the implementation.